MtfhReportingDataListener/Gateway/GlueGateway.cs

```diff
@@ -0,0 +1,71 @@
+using Amazon.Glue;
+using MtfhReportingDataListener.Gateway.Interfaces;
+using System.Threading.Tasks;
+using Amazon.Glue.Model;
+
+namespace MtfhReportingDataListener.Gateway
+{
+    public class GlueGateway : IGlueGateway
+    {
+        public IAmazonGlue _amazonGlueClient;
+        public GlueGateway(IAmazonGlue amazonGlueClient)
+        {
+            _amazonGlueClient = amazonGlueClient;
+        }
+
+        public async Task<SchemaResponse> GetSchema(string registryName, string schemaArn, string schemaName)
+        {
+            var latestVersionNumber = await GetLastestSchemaVersionNumber(registryName, schemaArn, schemaName).ConfigureAwait(false);
+            var SchemaDefinition = await GetSchemaDefinition(registryName, schemaArn, schemaName, latestVersionNumber).ConfigureAwait(false);
+
+            return new SchemaResponse
+            {
+                Schema = SchemaDefinition,
+                VersionId = latestVersionNumber
+            };
+        }
+
+        private async Task<string> GetSchemaDefinition(string registryName, string schemaArn, string schemaName, long versionNumber)
+        {
+            var schemaVersionRequest = new GetSchemaVersionRequest()
+            {
+                SchemaId = new SchemaId()
+                {
+
+                    RegistryName = registryName,
+                    SchemaArn = schemaArn,
+                    SchemaName = schemaName
+
+                },
+                SchemaVersionNumber = new SchemaVersionNumber()
+                {
+                    LatestVersion = true,
```

Comment:
Would we always want the latest version or would we need to reconfigure this in the future if for example we are supporting multiple API versions with differences in data model?

Reply:
This will be part of the improvements that the data platform team will work on next :)